pluginops/prs.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/go-github/v32/github"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+)
+
+func init() {
+	rootCmd.AddCommand(prsCmd)
+
+	prsCmd.AddCommand(
+		prsListCmd,
+		prsMergeCmd,
+	)
+}
+
+var prsCmd = &cobra.Command{
+	Use:   "prs",
+	Short: "Manage PRs.",
+}
+
+var prsListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List your plugin PRs.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := getGitHubClient()
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		prs, err := getPluginPRs(ctx, client)
+		if err != nil {
+			return err
+		}
+
+		slices.SortFunc(prs, func(a, b *prInfo) bool { return a.Repository < b.Repository })
+
+		for _, pr := range prs {
+			out := fmt.Sprintf("%s: %s", pr.Repository, pr.GetTitle())
+
+			if pr.readyToBeMerged() {
+				out += Green("\u2713")
+			} else {
+				out += Red("\u2715")
+			}
+			fmt.Print(out + "\n")
+		}
+
+		return nil
+	},
+}
+
+var prsMergeCmd = &cobra.Command{
+	Use:   "merge",
+	Short: "Merge plugin PRs that are ready.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := getGitHubClient()
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		prs, err := getPluginPRs(ctx, client)
+		if err != nil {
+			return err
+		}
+
+		if len(prs) == 0 {
+			fmt.Println("No PRs to merge.")
+			return nil
+		}
+
+		slices.SortFunc(prs, func(a, b *prInfo) bool { return a.Repository < b.Repository })
+
+		for _, pr := range prs {
+			if !pr.readyToBeMerged() {
+				log.WithFields(log.Fields{"repo": pr.Repository, "title": pr.GetTitle()}).Debug("PR not ready to be merged")
+				continue
+			}
+
+			ok, err := confirmPrompt(fmt.Sprintf("Merge %s/%s", pr.Repository, pr.GetTitle()))
+			if err != nil {
+				return err
+			}
+			if !ok {
+				continue
+			}
+
+			err = mergePR(context.Background(), client, pr)
+			if err != nil {
+				return err
+			}
```

Comment:
For these errors, should we log and `continue`, instead of returning? If so, maybe put this loop body in a separate function so we easily log errors and move on. Maybe that extraction wouldn't provide much value though

Reply:
Logging the error sound good to me. `mergePR` is already an abstraction. I don't think an additionl method makes sense here.